pkg/authenticator/authenticator.go

```diff
@@ -0,0 +1,19 @@
+package authenticator
```

Comment:
Minor thing, but this filename isn't great, as it's too generic. What is it authenticating to?